File: lab2/Lab2.java
package lab2;

class Point {

    int x;
    int y;

    public Point(int x, int y){
        this.x = x;
        this.y = y;
    }

    public Point reflect_x(){
        int newX = this.x;
        int newY = this.y * -1;

        Point newPoint = new Point(newX, newY);
        return newPoint;
    }

    public String get_line_to(Point endPoint){
        int x1 = this.x;
        int x2 = endPoint.x;
        int y1 = this.y;
        int y2 = endPoint.y;
        int m = (y2 - y1)/(x2 - x1);
        int b = (y1 - (m*x1));
        return "y =" + m + "x + " +b;
    }

}
class Main{
    public static void main(String[] args) {
        Point first = new Point(3, -5);
        System.out.printf("First (%d, %d)", first.x, first.y);
        System.out.printf(" Second (%d, %d)", first.reflect_x().x, first.reflect_x().y);
        System.out.println(" ");
        System.out.println(new Point(4, 11).get_line_to(new Point(6, 15)));

    }
}

File: lab2/Lab2.py


class Point:
    def __init__(self, x, y):
        self.x = x
        self.y = y

    def reflect_x(self):
        newX = self.x
        newY = self.y * -1
        return Point(newX, newY)

    def get_line_to(self, endPoint):
        x1 = self.x
        x2 = endPoint.x
        y1 = self.y
        y2 = endPoint.y
        
        m = (y2 - y1)/(x2 - x1)
        
        b = (y1- (m*x1))

        #print(b)
        
        
        return ("y =" + str(m) + "x + " + str(b))

def main():


    first = Point(3, -5)
    second = first.reflect_x()
    #### Exercise 2
    print("First (" + str(first.x) + ", " + str(first.y) + ") Second (" + str(second.x) + ", " + str(second.y) + ")")
    ### Exercise 4
    print(Point(4, 11).get_line_to(Point(6, 15)))

main()
